floris/parallel_floris_model_2.py

```diff
@@ -0,0 +1,309 @@
+from __future__ import annotations
+
+import copy
+import warnings
+from pathlib import Path
+from time import perf_counter as timerpc
+
+import numpy as np
+import pandas as pd
+
+from floris.core import State
+from floris.floris_model import FlorisModel
+from floris.optimization.yaw_optimization.yaw_optimizer_sr import YawOptimizationSR
+
+
+class ParallelFlorisModel(FlorisModel):
+    """
+    This class mimics the FlorisModel, but enables parallelization of the main
+    computational effort.
+    """
+
+    def __init__(
+        self,
+        configuration: dict | str | Path | FlorisModel,
```

Comment:
Is the FlorisModel type hint consistent with the fact that it will give a warning if passed in?

Reply:
Good point---I've removed `FlorisModel` from the list, but checked that it does still run if an instantiated `FlorisModel` is passed in.